src/dbus/dbus_message_helper.cpp

```diff
@@ -27,17 +27,32 @@
  */
 
 #include "dbus/dbus_message_helper.hpp"
+#include "dbus/constants.hpp"
 
 namespace otbr {
 namespace dbus {
 
 otbrError DBusMessageExtract(DBusMessageIter *aIter, bool &aValue)
+{
+    otbrError   err = OTBR_ERROR_DBUS;
+    dbus_bool_t val;
+
+    VerifyOrExit(dbus_message_iter_get_arg_type(aIter) == DBUS_TYPE_BOOLEAN);
+    dbus_message_iter_get_basic(aIter, &val);
+    dbus_message_iter_next(aIter);
+    aValue = (val != 0);
+    err    = OTBR_ERROR_NONE;
+exit:
+    return err;
+}
+
+otbrError DBusMessageExtract(DBusMessageIter *aIter, int8_t &aValue)
 {
     otbrError err = OTBR_ERROR_NONE;
     uint8_t   val;
 
     SuccessOrExit(err = DBusMessageExtract(aIter, val));
-    aValue = (val != 0);
+    aValue = static_cast<int8_t>(aValue);
```

Comment:
`aValue` is already `int8_t`. Casting is not needed.

Reply:
```suggestion
    aValue = static_cast<int8_t>(val);
```
Maybe you mean this?